### ISSUE TITLE
value_input_option='USER_ENTERED' not functioning for worksheet.insert_row()

### ISSUE THREAD

**Issue by bonzai299** [burnash/gspread#885]
**Important**: Please do not post usage questions here. 
To get a quick response, please ask a question on Stack Overflow using `gspread` tag.
See existing questions: https://stackoverflow.com/questions/tagged/gspread

---

**Describe the bug**
value_input_option='USER_ENTERED' parameter does not take effect for worksheet.insert_row() as it does for worksheet.append_row().

**To Reproduce**
1. define formula inside of fields variable
2. insert row using field variable and 'USER_ENTERED' parameter
3. append row using field variable and 'USER_ENTERED' parameter
4. inserted row shows '=SUM(L2:O2) as cell value
5. appended row shows =SUM(L2:O2) as cell value (rendered properly as formula)

**Expected behavior**
Row inserted at index 2 and row appended to end of sheet should both accurately respect the 'USER_ENTERED' parameter and result in no ' at the beginning of cell values.

**Code example***
```
fields = [product, "Pending Shipment", msgtimestamp,'', accountemail,'','',"=SUM(L2:O2)",'','','','','','','','', itemprice,'']
worksheet.insert_row(fields,index=2,value_input_option='USER_ENTERED')
worksheet.append_row(fields,value_input_option='USER_ENTERED')
```

**Screenshots**

**Environment info:**
- Operating System [e.g. Linux, Windows, macOS]: Windows
- Python version: 3.9
- gspread version: 3.7

**Stack trace or other output that would be helpful**
No errors presented. even when parameter is spelled incorrectly. (append row shows errors when spelled incorrectly)

**Additional context**

**Comment by thlintw** [comment]
I'm having this exact problem and somehow calling `insert_rows()` instead solved this. Which is weird though because it seems that `insert_row()` is just calling `insert_rows()` with a single row and other args are just passed down.

**Comment by lavigne958** [comment]
Hi @bonzai299 , thank you for reporting this issue.

This is strange because the 2 methods use the same API URL to proceed, the only difference is the use of the parameter `InsertDataOption` which is specified in in the request for `append_rows` but not for `insert_rows`.

I will look at it when I can.

**Comment by lavigne958** [comment]
I confirm the bug is here in version 3.7.0

I just ran a test on the master branch and it is fixed. It has been fixed by #873 .

Next release will include the fix.

Already fixed, coming in next release, closing this issue.
